### ISSUE TITLE
sporadic travis failures

### ISSUE THREAD

**Issue by boneskull** [mcollina/node-coap#77]
I just modified `README.md`, which triggered a build.  The build [failed](https://travis-ci.org/mcollina/node-coap/jobs/73797017#L403).

The issue is:

```
  1) server with a confirmable message should retry sending the response if it does not receive an ack four times before 45s:
      Uncaught AssertionError: expected 4 to deeply equal 5
      + expected - actual
      +5
      -4
      
      at /home/travis/build/mcollina/node-coap/test/server.js:624:29
      at Object.sinon.clock.callTimer (/home/travis/build/mcollina/node-coap/node_modules/sinon/lib/sinon/util/fake_timers.js:200:32)
      at Object.tick (/home/travis/build/mcollina/node-coap/node_modules/sinon/lib/sinon/util/fake_timers.js:145:28)
      at fastForward (/home/travis/build/mcollina/node-coap/test/server.js:51:11)
      at processImmediate [as _immediateCallback] (timers.js:363:15)
1``

I imagine this is just a timing issue, but I'm wondering if it's indicative of a code problem or simply a test problem.

**Comment by mcollina** [comment]
I have been seeing this for quite some time, but had no time to
investigate. It's a time related test, which for their nature tend to have
false positives. I fear the latter case, anyway.
Il giorno dom 2 ago 2015 alle 21:09 Christopher Hiller <
notifications@github.com> ha scritto:

**Comment by neophob** [comment]
just a wild guess: the tests use increasing udp ports - depending on the travis machine such a port could be in use, this the test fail? maybe use a fixed port?

**Comment by boneskull** [comment]
it may be worth trying [get-port](https://npmjs.com/package/get-port).  if that doesn't work, there's always the `retry` option in `.travis.yml`

**Comment by neophob** [comment]
I think the cleanest solution would be mocking for all udp requests, when we wouldn't have any issues anymore. Adding more logic to the test setup is a waste of time in my eyes. What's your view on this issue?

**Comment by mcollina** [comment]
get-port works for tcp.
The actual solution should be to use https://nodejs.org/api/dgram.html#dgram_socket_bind_port_address_callback without a fixed port (maybe 0 will do), and rewrite the tests to use that auto-discovered port instead.

**Comment by neophob** [comment]
can be closed, fixed with issue #104
